lib/Doctrine/ODM/PHPCR/DocumentRepository.php

```diff
@@ -168,7 +168,14 @@ public function findBy(array $criteria, array $orderBy = null, $limit = null, $o
                 $where = $qb->andWhere();
             }
 
-            $this->constraintField($where, $field, $value, 'a');
+            if (!is_array($value)) {
```

Comment:
for code readability, i would prefer to switch the cases, and have `if (is_array($value))`.